### ISSUE TITLE
How to config it?

### ISSUE THREAD

**Issue by rigzbe** [grazzolini/mkinitcpio-chkcryptoboot#7]
Hi :> 

I cant figure out what is need to paste to
CMDLINE_NAME= and CMDLINE_VALUE=

I added `123` after `CMDLINE_NAME=`
and a sha512sum after `CMDLINE_VALUE=`

and then i added to /etc/default/grub line 
`123=sha512sum`

is that good config?

**Comment by grazzolini** [comment]
I suggest you use two hashes, but hey, that's up to you. The only thing the script does is to see if the value you setup in the config file, is present on the kernel cmdline. Since on x86 the cmdline can have up to 4096 bytes, you can get away with two sha512sum hashes. Don't forget to recreate your grub config with grub-mkconfig.

**Comment by rigzbe** [comment]
I dont understood.

File '/etc/default/chkcryptoboot.conf' looks:
`...`
`CMDLINE_NAME=firstSum`
`CMDLINE_VALUE=secondSum`

file '/etc/default/grub' looks:
`...`
`GRUB_CMDLINE_LINUX_DEFAULT="quiet firstSum=secondSum"`

I recreated kernel using:
`mkinitcpio -p linuxxxx`
then I recreated grub config file:
`grub-mkconfig -o /boot/grub/grub.cfg`

after that I rebooted and I got alert from chkcryptoboothook :cry: 

Could You provide sample of 
' /etc/default/chkcryptoboot.conf' and  '/etc/default/grub' ?

**Comment by grazzolini** [comment]
Everything seems right. You can check everything by doing this:
1) Check if your /boot/grub/grub.cfg generated file indeed has the sums on the entries
2) You can extract your initrd using lsinitcpio -x, and you can confirm that your /etc/default/chkcryptoboot.conf file is correct
3) You can boot your machine passing break=premount to the initrd, so it will drop to a shell before mounting your root partition (you'll still need to type the password before this happens). Using that shell, you can inspect your /proc/cmdline file to see if it has the right parameters.

Also, which warning are you getting when you're booting. The warnings for missing kernel parameters are different than warnings for boot loader tampering detection.

**Comment by rigzbe** [comment]
Warning - default warning when sums are broken/missed `WARNING !!! Type uppercase yes to continue`

soo i checked
and
ad. 1.
in /boot/grub/grub.cfg `secondSum` one character was missing ;_;

ad. 2. `sudo lsinitcpio -x /boot/initramfs-3.18-x86_64.img` outputs nothing at all ;_;

ad. 3. I did not checked yet because after fixing missing character everything with chkcryptoboot works :smiley: 

`but after that when I'm loging to my user account from lightdm I got message  `Could not update ICEauthority file` ;_; I can login but with ugly window showing this message :sob: 
I tried owning this file but nothing changes.
When I type `iceauth` it says:
`iceauth:  timeout in locking authority file /home/rigzbe/.ICEauthority``

Funfact - somehow my $USER catalog was owned by root:root :smile:  I fixed that and (for now) ev works great. Much fanks for help : - }
